### PR TITLE
Task 7 - add Player class

### LINKED ISSUE #9: Task 7.1  - Player (class)
Napisz klasę Player, która posiadać będzie 3 pola:

- std::shared_ptr<Ship> ship_ (dla chętnych, spróbuj użyć std::unique_ptr<>)
- money_
- availableSpace_

Dopisz także odpowiednie gettery oraz konstruktor.

### LINKED ISSUE #10: Task 7.2 - Player (function getSpeed)
Dopisz funkcję, która powinna zwracać dane ze statku:

size_t getSpeed() const

### LINKED ISSUE #11: Task 7.3 - Player (function getCargo)
Dopisz funkcję, która powinna zwracać dane ze statku:

Cargo* getCargo(size_t index) const

### PR DESCRIPTION
**Pull request by m-kop**

### CODE REVIEW

File: shm/player.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#include <memory>
+
+#include "ship.hpp"
+
+class Player {
+public:
+    Player(Ship& ship, size_t money, size_t availableSpace);
+
+    std::unique_ptr<Ship> getShip() { return std::move(ship_); }
```
Comment (mariuszlisowski) [inline shm/player.hpp:11]: Do we need 'getShip' method?

File: shm/player.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    size_t getMoney() const { return money_; }
+    size_t getAvailableSpace() const { return availableSpace_; }
+
```
Comment (mariuszlisowski) [inline shm/player.hpp:16]: We also need the following methods:
size_t getSpeed() const;
Cargo* getCargo(size_t index) const;

File: shm/player.cpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include "player.hpp"
+
+Player::Player(Ship& ship, size_t money, size_t availableSpace)
+    : money_(money)
+    , availableSpace_(availableSpace) {
+    ship_ = std::make_unique<Ship>(ship);
```
Comment (mariuszlisowski) [inline shm/player.cpp:6]: Example implementation of 'getSpeed' method:
size_t Player::getSpeed() const { return this->ship_->getSpeed(); }

Example implementation of 'getCargo' method:
Cargo* Player::getCargo(size_t index) const { return ship_->getCargo(index); } **

** Note that get 'getCargo' method has to be implemented in a Ship class as well.

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

File: shm/player.cpp [hunk 4]
```diff
@@ -5,3 +5,7 @@ Player::Player(Ship& ship, size_t money, size_t availableSpace)
     , availableSpace_(availableSpace) {
     ship_ = std::make_unique<Ship>(ship);
 }
+
+std::shared_ptr<Cargo> Player::getCargo(size_t index) const {
+    return ship_->getCargo().get();
+}
```
Comment (mariuszlisowski) [inline shm/player.cpp:11]: Please use getCargo(const size_t index) method form Ship class, i.e.  ship_->getCargo(index);

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Review by mariuszlisowski — APPROVED** [review]

**Comment by m-kop** [comment]
I going to update this PR later on

File: shm/player.hpp (hunk 1 above)
Comment (m-kop) [reply shm/player.hpp:11]: we were asked to provide getters to members so it's one of them

**Pull request merged**